### PR TITLE
Fixes #859, update of #860

### LINKED ISSUE #859: annotate("segment", ...) doesn't work with dates on x-axis
How to reproduce:

``` r
library(data.table)
library(lubridate)
library(ggplot2)

# Prepare some random data
set.seed(1234)
dt <- data.table(x = rnorm(365*5), d = seq(ymd(20130101), ymd(20131231), by = 86400))
dt.m <- dt[, list(total = sum(x)), by = list(month = floor_date(d, "month"))]
# Create a basic scatterplot chart
p <- qplot(month, total, data = dt.m)
```

This works:

``` r
# Both of these work as expected and produce the same result
p + geom_segment(x = as.numeric(ymd(20130401)), xend = as.numeric(ymd(20130701)), 
  y = -10, yend = 10)
p + geom_segment(aes(x = ymd(20130401), xend = ymd(20130701), 
  y = -10, yend = 10))
```

but this doesn't:

``` r
> p + annotate("segment", x = as.numeric(ymd(20130401)), xend = as.numeric(ymd(20130701)), 
  y = -10, yend = 10)
Error: Invalid input: time_trans works with objects of class POSIXct only

> p + annotate("segment", x = ymd(20130401), xend = ymd(20130701), 
  y = -10, yend = 10)
Error in Ops.POSIXt((x - from[1]), diff(from)) : 
  '/' not defined for "POSIXt" objects 

> p + annotate("segment", aes(x = ymd(20130401), xend = ymd(20130701), 
  y = -10, yend = 10))
Error in as.data.frame.default(x[[i]], optional = TRUE, stringsAsFactors = stringsAsFactors) : 
  cannot coerce class ""uneval"" to a data.frame
```

I have also posted a question to SO - there may be some additional details there. http://stackoverflow.com/questions/18796386/how-to-use-ggplot2s-annotate-with-dates-in-x-axis

### PR DESCRIPTION
**Pull request by BrianDiggs**
Fixes bug where `xend` and `yend` were not passed along
as coordinates for transformation within `annotate`. This
led to the end of segments not appearing in the correct
location when a transformation was involved with one of the
coordinate scales.

### CODE REVIEW

**Comment by hadley** [comment]
Similarly, could you please merge/rebase for NEWS? (it's safest to always put new NEWS items at the top)

File: inst/tests/test-annotate.r [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  p <- qplot(month, total, data = dt)
+
+  ggplot_gtable(ggplot_build(p + annotate("segment",
+               x = as.POSIXct("2013-04-01"),
+               xend = as.POSIXct("2013-07-01"),
+               y = -10,
+               yend = 10)))
+
+})
```
Comment (hadley) [inline inst/tests/test-annotate.r]: These don't assert anything?

File: inst/tests/test-annotate.r [hunk 2]
```diff
@@ -0,0 +1,31 @@
+context("annotate")
+
+test_that("dates in segment annotation work", {
+  dt <-
```
Comment (hadley) [inline inst/tests/test-annotate.r]: Could you please simplify this dataset to just use a couple of dates created by hand?

**Comment by hadley** [comment]
Are you still interested in this?

**Comment by BrianDiggs** [comment]
I am still interested in this. I'll work on it this weekend.

**Comment by BrianDiggs** [comment]
Comments addressed.

**Comment by hadley** [comment]
Thanks!

**Pull request merged**